source/Ship.cpp

```diff
@@ -125,6 +125,8 @@ void Ship::Load(const DataNode &node)
 				armament.AddGunPort(hardpoint, outfit);
 			else
 				armament.AddTurret(hardpoint, outfit);
+			if (armament.Get().size() > 32)
+				child.PrintTrace("Warning: Ships with more than 32 hardpoints do not behave as one would expect:");
```

Comment:
This would print out an error message for *every* hardpoint after the first 32, which could be dozens of lines of errors. It would make more sense to just do the check once at the end of Ship::Load() and only print one message per ship.

Also, your warning message is rather vague. It would be simpler to just say, "Warning: ship has more than 32 weapon hardpoints. Some weapons may not fire:"